File: UI/wwwroot/SystemDocuments/Models/Article.js
//@ts-check
import { EntityClass } from "../../WDevCore/WModules/EntityClass.js";
import { Category } from "./Category.js";

class Article extends EntityClass {
    /** @param {Partial<Article>} [props] */
    constructor(props) {
        super(props, 'ArticleManager');
        for (const prop in props) {
            this[prop] = props[prop];
        }
    }
    /** @type {Number} */ Article_Id;
    /** @type {String} */ Title;
    /** @type {String} */ Author;
    /** @type {Number} */ Id_User;
    /** @type {String} */ Body;
    /** @type {Date} */ Publish_Date;
    /** @type {Date} */ Update_Date;
    /** @type {Boolean} */ Status;
    /** @type {Number} */ Category_Id;
    /** @type {Category} ManyToOne */ Category;
}

export { Article };

File: BusinessLogic/SystemDocuments/Models/Article.cs
using System;
using APPCORE;

namespace BusinessLogic.SystemDocuments.Models
{
	public class Article : EntityClass
	{
		[PrimaryKey(Identity = true)]
		public int? Article_Id { get; set; }
		public string? Title { get; set; }
		public string? Author { get; set; }		
		public int? Id_User { get; set; }
		public string? Body { get; set; }
		public DateTime? Publish_Date { get; set; }
		public DateTime? Update_Date { get; set; }
		public bool? Status { get; set; }
		// Foreign key
		public int? Category_Id { get; set; }
		// Navigation property
		[ManyToOne(TableName = "Category", KeyColumn = "Category_Id", ForeignKeyColumn = "Category_Id")]
		public Category? Category { get; set; }
	}
}